source/agora/node/admin/AdminInterface.d

```diff
@@ -183,9 +189,10 @@ public class AdminInterface : NodeControlAPI
     public override string encryptionKeyQR (
         string app, ulong height, out string contentType, out string vary)
     {
-        // TODO: PreImage to be released at that height
-        // Or it can be pre-released, so it can be `height + PreimageRevealPeriod`
-        PreImageInfo pre_image;
+        // Gets the pre-image of the requested height.
+        // It can be pre-released, so it add to the `Reveal period + 1`
+        Hash pre_image = this.enroll_man.getOurPreimage(
+            Height(height + this.enroll_man.PreimageRevealPeriod + 1));
```

Comment:
No it should just be `height`, the app requesting it should compensate for `PreImageRevealPeriod`

Reply:
Okay, I updated.